doc/source/ray-core/ray-generator.rst

```diff
@@ -0,0 +1,210 @@
+.. _generators:
+
+Ray Generators 
+==============
+
+`Python generators <https://docs.python.org/3/howto/functional.html#generators>`_ are functions
+that behave like an iterator, yielding one value per iteration. Ray also supports the generators API.
+
+.. code-block:: diff
+
+    +import ray
+     import time
+
+     # Takes 25 seconds to finish.
+    +@ray.remote(num_returns="streaming")
+     def f():
+         for i in range(5):
+             time.sleep(5)
+             yield i
+
+    -for obj in f():
+    +for obj_ref in f.remote():
+         # It is printed every 5 seconds, not after 25 seconds.
+    -    print(obj)
+    +    print(ray.get(obj_ref))
+
+
+The above Ray generator yields the output every 5 seconds 5 times.
+When you use a normal Ray task, you have to wait 25 seconds to access the output. 
+With a Ray generator, the caller can access the object refeference
+before the task ``f`` finishes.
+
+**Ray generator is useful when**
```

Comment:
We should also talk about when this is not preferred?

Basically help user decide whether to use static num_returns or streaming num_returns.

Reply:
Hmm should we add a new section? I feel like it should be a pattern / anti-pattern doc rather than this doc?